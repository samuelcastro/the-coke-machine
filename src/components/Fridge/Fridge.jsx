import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Display, Panel } from '../'

class Fridge extends PureComponent {
  render() {
    const {
      coins,
      products,
      message,
      title,
      onPayDispatch,
      onChooseDispatch,
      onRemoveDispatch,
      onReturnCoinDispatch
    } = this.props;

    return (
      <React.Fragment>
        <h1>{title}</h1>

        <Display value={message}/>

        <Panel
          coins={coins}
          products={products}
          onPay={onPayDispatch}
          onChoose={onChooseDispatch}
          onRemove={onRemoveDispatch}
          onReturnCoin={onReturnCoinDispatch}
        />
      </React.Fragment>
    );
  }
}

Fridge.propTypes = {
  coins: PropTypes.array.isRequired,
  products: PropTypes.array.isRequired,
  message: PropTypes.any.isRequired,
  title: PropTypes.string.isRequired,
  onPayDispatch: PropTypes.func.isRequired,
  onChooseDispatch: PropTypes.func.isRequired,
  onRemoveDispatch: PropTypes.func.isRequired,
  onReturnCoinDispatch: PropTypes.func.isRequired
}

export default Fridge;
