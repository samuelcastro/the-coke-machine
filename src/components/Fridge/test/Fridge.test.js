import React from 'react';
import renderer from 'react-test-renderer';
import Fridge from '../Fridge';

describe('Rendering Fridge', () => {
  let fridge, wrapper;

  const mockedTitle = 'The Coke Machine';
  const mockedMessage = 'Welcome';
  const mockedCoins = [0.25, 0.10, 0.05];
  const mockedProducts = [{
    id: 1,
    name: 'Coke',
    quantity: 5
  }, {
    id: 2,
    name: 'Diet Coke',
    quantity: 5
  }]

  beforeEach(() => {
    fridge = <Fridge
      title={mockedTitle}
      coins={mockedCoins}
      products={mockedProducts}
      message={mockedMessage}
      onPayDispatch={jest.fn}
      onChooseDispatch={jest.fn}
      onRemoveDispatch={jest.fn}
      onReturnCoinDispatch={jest.fn}
    />

    wrapper = mount(fridge);
  });

  it('should render <Fridge /> correctly', () => {
    const tree = renderer
      .create(fridge)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain title', () => {
    expect(wrapper.contains(<h1>{mockedTitle}</h1>)).toEqual(true);
  });

  it('should contain a Display component', () => {
    expect(wrapper.find('Display').length).toEqual(1);
  });

  it('should contain a Panel component', () => {
    expect(wrapper.find('Panel').length).toEqual(1);
  });
});
