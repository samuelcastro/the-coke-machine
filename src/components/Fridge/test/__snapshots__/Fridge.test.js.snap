// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Fridge should render <Fridge /> correctly 1`] = `
Array [
  <h1>
    The Coke Machine
</h1>,
  <section>
    Welcome
</section>,
  <section
    className="coins"
>
    <button
        onClick={[Function]}
    >
        Insert 
        0.25
    </button>
    <button
        onClick={[Function]}
    >
        Insert 
        0.1
    </button>
    <button
        onClick={[Function]}
    >
        Insert 
        0.05
    </button>
</section>,
  <section
    className="products"
>
    <button
        onClick={[Function]}
    >
        Coke
    </button>
    <button
        onClick={[Function]}
    >
        Diet Coke
    </button>
</section>,
  <section
    className="take"
>
    <button
        onClick={[Function]}
    >
        Remove Product
    </button>
</section>,
  <section
    className="coin-return"
>
    <button
        onClick={[Function]}
    >
        Coin Return
    </button>
</section>,
]
`;
